src/main/java/org/embulk/input/zendesk/services/ZendeskCustomObjectService.java

```diff
@@ -0,0 +1,109 @@
+package org.embulk.input.zendesk.services;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.http.HttpStatus;
+import org.embulk.config.ConfigException;
+import org.embulk.config.TaskReport;
+import org.embulk.input.zendesk.RecordImporter;
+import org.embulk.input.zendesk.ZendeskInputPlugin;
+import org.embulk.input.zendesk.clients.ZendeskRestClient;
+import org.embulk.input.zendesk.models.Target;
+import org.embulk.input.zendesk.models.ZendeskException;
+import org.embulk.input.zendesk.stream.paginator.sunshine.CustomObjectSpliterator;
+import org.embulk.input.zendesk.utils.ZendeskConstants;
+import org.embulk.input.zendesk.utils.ZendeskUtils;
+import org.embulk.spi.Exec;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class ZendeskCustomObjectService implements ZendeskService
+{
+    protected ZendeskInputPlugin.PluginTask task;
+
+    private ZendeskRestClient zendeskRestClient;
+
+    public ZendeskCustomObjectService(final ZendeskInputPlugin.PluginTask task)
+    {
+        this.task = task;
+    }
+
+    @VisibleForTesting
+    public ZendeskRestClient getZendeskRestClient()
+    {
+        if (zendeskRestClient == null) {
+            zendeskRestClient = new ZendeskRestClient();
+        }
+        return zendeskRestClient;
+    }
+
+    public boolean isSupportIncremental()
+    {
+        return false;
+    }
+
+    @Override
+    public TaskReport execute(final int taskIndex, final RecordImporter recordImporter)
+    {
+        final List<String> paths = getListPathByTarget();
+
+        paths.parallelStream().forEach(path -> StreamSupport.stream(new CustomObjectSpliterator(path, getZendeskRestClient(), task, Exec.isPreview()), Exec.isPreview())
+                .forEach(recordImporter::addRecord));
+
+        return Exec.newTaskReport();
+    }
+
+    @Override
+    public JsonNode getData(final String path, final int page, final boolean isPreview, final long startTime)
+    {
+        Optional<String> response = Optional.empty();
+
+        final List<String> paths = path.isEmpty()
+                ? getListPathByTarget()
+                : Collections.singletonList(path);
+
+        for (final String temp : paths) {
+            try {
+                response = Optional.ofNullable(getZendeskRestClient().doGet(temp, task, true));
+
+                // in guessing, break when we have data
+                if (response.isPresent()) {
```

Comment:
fixed

Reply:
Actually you just update the comment sounds not enough. My point here `getDataFromPath` is assumed only called during `preview` or `guess` process which knows only when dive into code, `isPreview` argument is actually useless.

I understood that `custom object` does not support incremental and `isPreview` must be always `true`. What if we do the `Precondition` check that `isPreview` must be true to execute further that method? Otherwise invoker in the future might call it with `false` value and they see another result.